### PR TITLE
Janus http client

### PR DESCRIPTION
**Pull request by vogdb**

### CODE REVIEW

**Comment by vogdb** [comment]
@tomaszdurka please review

**Comment by vogdb** [comment]
@tomaszdurka please review

File: lib/janus-http-client.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+var requestPromise = require('request-promise');
+var serviceLocator = require('./service-locator');
+var ProxyConnection = require('./proxy-connection');
+
+function JanusHttpClient(url) {
+  this.baseUrl = url;
+}
+
+JanusHttpClient.prototype._httpClient = function() {
```
Comment (tomaszdurka) [inline lib/janus-http-client.js]: This method is a bit strange, same as in CmApiClient (where it was copied from?).

Why not proxy whole method aka
```
JanusHttpClient.prototype._httpRequest = requestPromise;
```
?

File: lib/janus-http-client.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    method: 'POST',
+    uri: this.baseUrl + action,
+    json: true
+  };
+  if (data) {
+    options.body = data;
+  }
+
+  serviceLocator.get('logger').info('janus-http-client', 'request', options.uri, options.body);
+  return this._httpClient()(options)
```
Comment (tomaszdurka) [inline lib/janus-http-client.js]: As mentioned above this is quite strange usage. We treat httpClient as method which returns method. Hmmm. Let's do as noted above or similiar.

**Comment by tomaszdurka** [comment]
Rest looks ok!

File: lib/janus-http-client.js (hunk 1 above)
Comment (fvovan) [inline lib/janus-http-client.js]: That was me, I used it to mock `request-promise`
as `requestPromise` is a function I just returned it

**Comment by tomaszdurka** [comment]
Pair-finished with @fvovan

**Pull request merged**